src/rpc/blockchain.cpp

```diff
@@ -680,21 +680,17 @@ static CBlock GetBlockChecked(BlockManager& blockman, const CBlockIndex& blockin
 
 static std::vector<std::byte> GetRawBlockChecked(BlockManager& blockman, const CBlockIndex& blockindex)
 {
-    std::vector<std::byte> data{};
     FlatFilePos pos{};
     {
         LOCK(cs_main);
         CheckBlockDataAvailability(blockman, blockindex, /*check_for_undo=*/false);
         pos = blockindex.GetBlockPos();
     }
 
-    if (!blockman.ReadRawBlock(data, pos)) {
-        // Block not found on disk. This shouldn't normally happen unless the block was
-        // pruned right after we released the lock above.
-        throw JSONRPCError(RPC_MISC_ERROR, "Block not found on disk");
-    }
-
-    return data;
+    if (auto data{blockman.ReadRawBlock(pos)}) return *data;
```

Comment:
nit
```suggestion
    if (const auto data{blockman.ReadRawBlock(pos)}) return *data;
```

Reply:
https://github.com/bitcoin/bitcoin/compare/bbea9636a3b11df1c79664e97ded7907bbacfcf3..9efccb0d3e84d83e0ec29a4b18b72019ffc3d5c3

Reply:
The compiler can't automatically move `const` `lvalue` return values to the callsite.

If this was just returning the value, like `return data;`, then it would be straightforward and `const` would definitely be undesirable and require a copy. Regardless, the compiler here won't be able to move the expected value out of the `Expected` struct if it is `const`.

Since we have to dereference here and are not returning the `lvalue` itself, I'm unsure if the compiler will be able to move the expected value out of the `Expected` or not. If not, we should wrap it in `std::move`.